src/pharmpy/modeling/expressions.py

```diff
@@ -864,22 +864,43 @@ def has_random_effect(model: Model, parameter: str, level: str = 'all') -> bool:
     return _depends_on_any_of(model, parameter, rvs)
 
 
-def get_rv_parameter(model: Model, rv: str) -> str:
-    # NOTE This was just copied here and is used elsewhere but should be made
-    # more general
-    sset = model.statements
+def get_rv_parameters(model: Model, rv: str) -> List[str]:
+    """Retrieves parameters in :class:`pharmpy.model` given a random variable.
+
+    Parameters
+    ----------
+    model : Model
+        Pharmpy model to retrieve parameters from
+    rv : str
+        Name of random variable to retrieve
+
+    Return
+    ------
+    Parameters
+        The parameters of the random variable for the given model
+
+    Example
+    -------
+    >>> from pharmpy.modeling import *
+    >>> model = load_example_model("pheno")
+    >>> get_rv_parameters(model, 'ETA(1)')
+    ['CL']
+
+    See also
+    --------
+    has_random_effect
+    get_pk_parameters
 
-    s = _find_assignment(sset, sympy.Symbol(rv))
+    """
+    if rv not in model.random_variables.names:
+        raise ValueError(f'Could not find random variable: {rv}')
 
-    if s is None:
-        raise ValueError(f'Could not find an assignment to {rv}')
+    natural_assignments = _get_natural_assignments(model.statements.before_odes)
 
-    if len(s.expression.free_symbols) > 1:
-        return s.symbol.name
-    else:
-        s = _find_assignment(sset, s.symbol)
-        assert s is not None
-        return s.symbol.name
+    free_symbols = model.statements.free_symbols
+    dependency_graph = _dependency_graph(natural_assignments)
+    filtered = list(_filter_symbols(dependency_graph, free_symbols, {sympy.Symbol(rv)}))
```

Comment:
Use `sorted` here to get predictable output.

Reply:
You can bypass the `list` construction: `sorted(_filter_symbols(...))`.

Reply:
@stellabelin ?

Reply:
Sorry, missed the last comment, updating

Reply:
Updated